### PR TITLE
385 adapter the taobaominigame ClearLRU procedure

### LINKED ISSUE #17040: taobao Platform can not trigger clearLRU
### Cocos Creator version

3.x

### System information

All Devices in Taobaominigam

### Issue description

taobao Platform can not trigger clearLRU.
in cache-manager function _cache,unzipAndCacheBundle, there is a mothed"isOutOfStorage" which will judge weather will trigger clearLRU, but in taobaominigame the right error  regular expression is "outOfStorageRegExp = /10028/; "
so in xxx\cocos\editors\Creator\2.4.12\resources\builtin\adapters\platforms\taobao-minigame\wrapper\fs-utils.js, the code var outOfStorageRegExp = /the maximum size of the file storage/;
will change to var outOfStorageRegExp = /10028/;

### Relevant error log output

_No response_

### Steps to reproduce

when cache is out of memory, will cause this issue

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by tangkaikk**
Re: #

### Changelog

*
Fixes https://github.com/cocos/cocos-engine/issues/17040 : taobao Platform can not trigger clearLRU
#17230 
-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

## Greptile Summary

- Added `cache-manager.js` for Taobao Mini Game platform
- Implemented unzip and cache functionality for bundles
- Added handling for out-of-storage scenarios
- Introduced cache file management and cleaning
- Adapted common cache manager for platform-specific needs

### CODE REVIEW

File: platforms/runtime/common/engine/cache-manager.js [hunk 1]
```diff
@@ -188,8 +193,12 @@ const cacheManager = {
         function deferredDelete () {
             const item = caches.pop();
             if (self._isZipFile(item.originUrl)) {
-                rmdirSync(item.url, true);
-                self._deleteFileCB();
+                if (self._isZipFile(item.url)) {
+                    deleteFile(item.url, self._deleteFileCB.bind(self));
+                } else {
+                    rmdirSync(item.url, true);
```
Comment (minggo) [inline platforms/runtime/common/engine/cache-manager.js]: Why need to sync?

File: platforms/runtime/common/engine/cache-manager.js [hunk 2]
```diff
@@ -188,8 +193,12 @@ const cacheManager = {
...
+                    self._deleteFileCB();
+                }
             } else {
                 deleteFile(item.url, self._deleteFileCB.bind(self));
```
Comment (minggo) [inline platforms/runtime/common/engine/cache-manager.js]: Why don't have to check dir here?

File: platforms/minigame/common/engine/cache-manager.js [hunk 3]
```diff
@@ -223,16 +214,20 @@ const cacheManager = {
...
-                } else {
-                    rmdirSync(path, true);
-                    this._deleteFileCB();
-                }
-            } else {
+            this._remove(url, path);
+        }
+    },
+
+    _remove (url, path) {
```
Comment (minggo) [inline platforms/minigame/common/engine/cache-manager.js]: Please use a more meaningful function name.

Comment (tangkaikk) [reply platforms/minigame/common/engine/cache-manager.js]: gotcha

**Review by minggo — APPROVED** [review]

File: platforms/minigame/common/engine/cache-manager.js [hunk 4]
```diff
@@ -22,7 +22,7 @@
  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  THE SOFTWARE.
  ****************************************************************************/
-const {
+ const {
```
Comment (dumganhar) [inline platforms/minigame/common/engine/cache-manager.js]: Why add a space here?

File: platforms/minigame/common/engine/cache-manager.js [hunk 5]
```diff
@@ -183,10 +183,11 @@ const cacheManager = {
             caches.push({ originUrl: key, url: val.url, lastTime: val.lastTime });
         });
         caches.sort((a, b) => a.lastTime - b.lastTime);
-        caches.length = Math.floor(caches.length / 3);
-        if (caches.length === 0) {
-            cleaning = false;
-            return;
+        // cache length above 3 then clear 1/3, or clear all caches
+        if (caches.length < 3) {
+            console.warn('Due to caching large files in the game, there is insufficient storage space. Now starting forced cleaning.');
```
Comment (dumganhar) [inline platforms/minigame/common/engine/cache-manager.js]: This warning message takes **106** bytes. Could it be shorter? For instance:

> Insufficent storage, cleaning

File: platforms/minigame/common/engine/cache-manager.js [hunk 6]
```diff
@@ -198,12 +199,7 @@ const cacheManager = {
         this._write();
         function deferredDelete () {
             const item = caches.pop();
-            if (self._isZipFile(item.originUrl)) {
-                rmdirSync(item.url, true);
-                self._deleteFileCB();
-            } else {
-                deleteFile(item.url, self._deleteFileCB.bind(self));
-            }
+            self._removePathOrFile(item.originUrl, item.url);
```
Comment (dumganhar) [inline platforms/minigame/common/engine/cache-manager.js:202]: Why call `item.originalUrl` `url` and call `item.url`  `path`?
And why is the method called `_removePathOrFile(url, path)`? The method name `_removePathOrFile` doesn't say any about `url`, and `PathOrFile`, `(url, path)` are mismatch. So `File` is for `url` argument or `path` argument? I guess you mean `path` argument, in that case, Is `File` for `url` argument? and the order ?

File: platforms/minigame/common/engine/cache-manager.js (hunk 5 above)
Comment (tangkaikk) [reply platforms/minigame/common/engine/cache-manager.js]: ok，i will try

**Pull request merged**